WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java

```diff
@@ -249,8 +252,25 @@ private void loggedInAndFinish(ArrayList<Integer> oldSitesIds, boolean doLoginUp
                 break;
             case SHARE_INTENT:
             case SELFHOSTED_ONLY:
+                // We are comparing list of site ID's before self-hosted site was added and after, trying to find a
+                // newly added self-hosted site's ID, so we can select it
+                ArrayList<Integer> currentSiteIds = new ArrayList<>();
+                int newlyAddedSiteId = -1;
+
+                for (SiteModel site : mSiteStore.getSites()) {
+                    currentSiteIds.add(site.getId());
+                }
+
+                currentSiteIds.removeAll(oldSitesIds);
+
+                if (currentSiteIds.size() == 1) {
+                    newlyAddedSiteId = currentSiteIds.get(0);
+                }
+
+                Intent intent = new Intent();
+                intent.putExtra(SitePickerActivity.KEY_LOCAL_ID, newlyAddedSiteId);
```

Comment:
What do you think about leaving out setting the extra if we didn't find the newly added selfhosted site? Feels weird setting the Extra if we don't really mean passing it. Do we need it passed?

While at it, it seems that we might be letting the error condition (no site found) go by unhandled. I'm thinking, how about we check for `currentSiteIds`'s size being greater than `0` and report an error message if it's `0`?

Something like this:
```suggestion
                if (currentSiteIds.size() == 0) {
                    Log.e(T.MAIN, "Couldn't detect newly added self-hosted site!");
                    ToastUtils.showToast(this, R.string.failed_selecting_added_site);
                } else {
                    intent.putExtra(SitePickerActivity.KEY_LOCAL_ID, currentSiteIds.get(0));
                }
```

Reply:
Sounds good! Only thing I don't agree on is checking if the length of `currentSiteIds` is greater then `0`. If it's not `1` we will not be able to reliably select a correct site.  I left the `1` check and added an actual number of new site ID`s we got into the error log.

Reply:
I'm cool with the error handling in 72f2f6d 👍

Allow me to expand on the list size check: I think that the assumption that the list will either be 0 or 1 long is not accurate. I think what when adding a self-hosted multi-site, the size will be the amount of sites the multi-site has when you log-in as the superadmin. You can check out the test multi-site from http://do.wpmt.co/ to verify.

In the case the new sites are more than one, I think it should be fine to just select the first one in the list. WDYT?

Reply:
TIL about multi-sites! Had no idea this existed. I was thinking primarily of the case where there might be more than one site, and selecting the first one might be a bit misleading. I actually tested this scenario and looks like it's not a problem (If the new site was made elsewhere, while you were at the "Add self-hosted site" screen, it will not appear in diff).

I made the adjustments, and ready for another pass 🙇

Reply:
Thanks for the updates @khaykov , I green-lighted and merged the PR!

> If the new site was made elsewhere, while you were at the "Add self-hosted site" screen, it will not appear in diff

Just to be sure I'm not missing some important flow, can you elaborate on the flow there? I'm not sure I'm following. Do you mean perhaps that another site might have been created in the multi-site while in the wizard?